cmake-modules/AzureTransportAdapters.cmake

```diff
@@ -44,14 +50,14 @@ elseif (UNIX)
 
   if (BUILD_TRANSPORT_CURL OR (NOT BUILD_TRANSPORT_CURL AND NOT BUILD_TRANSPORT_CUSTOM))
 
-    if(NOT BUILD_TRANSPORT_CURL)
+    if(NOT BUILD_TRANSPORT_CURL AND NOT NO_AUTOMATIC_TRANSPORT_BUILD)
       message("No transport adapter was selected, using libcurl as the default option for POSIX.")
     endif()
 
-    add_compile_definitions(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
-
-    
-    SET(BUILD_TRANSPORT_CURL ON)
+    if (BUILD_TRANSPORT_CURL OR NOT NO_AUTOMATIC_TRANSPORT_BUILD)
+      add_compile_definitions(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
+      SET(BUILD_TRANSPORT_CURL ON)
+    endif()
```

Comment:
```suggestion
    if(NOT BUILD_TRANSPORT_CURL AND NOT NO_AUTOMATIC_TRANSPORT_BUILD)
      message("No transport adapter was selected, using libcurl as the default option for POSIX.")
      set(BUILD_TRANSPORT_CURL ON)
    endif()
    if (BUILD_TRANSPORT_CURL)
      add_compile_definitions(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
    endif()
```

Reply:
The displayed diff is wired

Reply:
I am going to rewrite the if-else branches here per Ahson's feedback, but the logic should remain the same, I'll tag you for the next PR iteration.

Reply:
@autoantwort, I re-arranged the condition. The diff still may look complicated, but essentially what I've done is that I wrote a simple and clear condition for the `if (NO_AUTOMATIC_TRANSPORT_BUILD)` case, and then pushed the previously existing logic without modifying (other than indentation) into an `else()`.